spec/integrations/mongo_spec.rb

```diff
@@ -0,0 +1,292 @@
+      end
+
+      it "is recursive and iterative for array values" do
+        expect(subscriber).to receive(:sanitize_filter_value).with([1, 2, 3], 0).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with(1, 1).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with(2, 1).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with(3, 1).ordered.and_call_original
+        expect(subscriber.send(:sanitize_filter_value, [1, 2, 3], 0)).to eq(['?', '?', '?'])
+      end
+
+      it "calls #santize_filter_hash for hash values" do
+        expect(subscriber).to receive(:sanitize_filter_hash).with({:a => 1}, 1)
+        subscriber.send(:sanitize_filter_value, {:a => 1}, 0)
+      end
+
+      it "increments the depth for each call" do
+        expect(subscriber).to receive(:sanitize_filter_value).with([1, [2, [3]]], 0).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with(1, 1).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with([2, [3]], 1).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with(2, 2).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with([3], 2).ordered.and_call_original
+        expect(subscriber).to receive(:sanitize_filter_value).with(3, 3).ordered.and_call_original
+        expect(subscriber.send(:sanitize_filter_value, [1, [2, [3]]], 0)).to eq(['?', ['?', ['?']]])
+      end
+
+      it "returns [MAX_FILTER_DEPTH_REACHED] if the filter depth is exceeded" do
+        expect(subscriber.send(:sanitize_filter_value, 1, 4)).to eq('[MAX_FILTER_DEPTH_REACHED]')
+      end
+    end
+
+    describe "#leave_command" do
+      it "extracts and stores the command by request_id" do
+        request_id = "123456"
+        command = "this is a command string"
+        event = double(:command => command, :request_id => request_id)
+
+        subscriber.send(:leave_command, event)
+        command_hash = Bugsnag.configuration.request_data[Bugsnag::MongoBreadcrumbSubscriber::MONGO_COMMAND_KEY]
+        expect(command_hash[request_id]).to eq(command)
+      end
+    end
+
+    describe "#pop_command" do
+      let(:request_id) { "123456" }
+      let(:command) { "this is a command string" }
+      before do
+        event = double(:command => command, :request_id => request_id)
+        subscriber.send(:leave_command, event)
+      end
+
+      it "returns the command given a request_id" do
+        expect(subscriber.send(:pop_command, request_id)).to eq(command)
+      end
+
+      it "removes the command from the request_data" do
+        subscriber.send(:pop_command, request_id)
+        command_hash = Bugsnag.configuration.request_data[Bugsnag::MongoBreadcrumbSubscriber::MONGO_COMMAND_KEY]
+        expect(command_hash).not_to include(request_id => command)
```

Comment:
I would change this to `not_to have_key(request_id)` as the key shouldn't be in at all and the value is not relevant to the test.